consoles/serial_screen.pm

```diff
@@ -166,20 +136,15 @@ sub do_read
     my $rin = '';
     vec($rin, fileno($fd), 1) = 1;
     my $nfound = select(my $rout = $rin, undef, my $eout = $rin, $args{timeout});
-    if ($nfound < 0) {
-        croak "Failed to select socket for reading: $ERRNO";
-    } elsif ($nfound == 0) {
-        return undef;
-    }
+    croak "Failed to select socket for reading: $ERRNO" if $nfound < 0;
+    return undef if $nfound == 0;
 
     my $read;
     while (!defined($read)) {
         $read = sysread($fd, $buffer, $args{max_size});
-        if (!defined($read) && !($ERRNO{EAGAIN} || $ERRNO{EWOULDBLOCK})) {
-            croak "Failed to read from virtio/svirt serial console char device: $ERRNO";
-        }
+        croak "Failed to read from virtio/svirt serial console char device: $ERRNO" if !defined($read) && !($ERRNO{EAGAIN} || $ERRNO{EWOULDBLOCK});
     }
-    $_[1] = $buffer;
+    $buf = $buffer;
```

Comment:
We use this called-by-reference, its equal to the `sysread(..., $buffer,...)` not sure how to do this with signatures.
```suggestion
    $_[1] = $buffer; 
```

Reply:
We could require one to specify a scalar ref here. That would make it more obvious that the variable *outside* the function is re-assigned.

Reply:
do you have an idea why this test would only fail within OBS and not GHA?

Reply:
Looks like some GHA glitch it failes: https://github.com/os-autoinst/os-autoinst/runs/4386959135?check_suite_focus=true#step:3:572

Reply:
Assigning to `$_[...]` willl still work with signatures, as `@_` is still working as before.

There is a dicussion on removing `@_`, though:
https://www.nntp.perl.org/group/perl.perl5.porters/2021/11/msg262052.html